pkg/proxmox/instances.go

```diff
@@ -131,85 +164,78 @@ func (i *instances) InstanceShutdown(ctx context.Context, node *v1.Node) (bool,
 func (i *instances) InstanceMetadata(ctx context.Context, node *v1.Node) (*cloudprovider.InstanceMetadata, error) {
 	klog.V(4).InfoS("instances.InstanceMetadata() called", "node", klog.KRef("", node.Name))
 
-	if providedIP, ok := node.ObjectMeta.Annotations[cloudproviderapi.AnnotationAlphaProvidedIPAddr]; ok {
-		var (
-			vmRef  *pxapi.VmRef
-			region string
-			err    error
-		)
-
-		providerID := node.Spec.ProviderID
-		if providerID == "" {
-			uuid := node.Status.NodeInfo.SystemUUID
+	var (
+		vmRef  *pxapi.VmRef
+		region string
+		err    error
+	)
 
-			klog.V(4).InfoS("instances.InstanceMetadata() empty providerID, trying find node", "node", klog.KObj(node), "uuid", uuid)
+	providerID := node.Spec.ProviderID
+	if providerID != "" && !strings.HasPrefix(node.Spec.ProviderID, util.ProviderName) {
+		klog.V(4).InfoS("instances.InstanceMetadata() omitting unmanaged node", "node", klog.KObj(node), "providerID", node.Spec.ProviderID)
 
-			mc := metrics.NewMetricContext("findVmByName")
-
-			vmRef, region, err = i.c.FindVMByNode(ctx, node)
-			if mc.ObserveRequest(err) != nil {
-				mc := metrics.NewMetricContext("findVmByUUID")
+		return &cloudprovider.InstanceMetadata{}, nil
+	}
 
-				vmRef, region, err = i.c.FindVMByUUID(ctx, uuid)
-				if mc.ObserveRequest(err) != nil {
-					return nil, fmt.Errorf("instances.InstanceMetadata() - failed to find instance by name/uuid %s: %v, skipped", node.Name, err)
-				}
-			}
+	if providerID == "" && util.HasTaintWithEffect(node, ccmapi.TaintExternalCloudProvider, "") {
+		uuid := node.Status.NodeInfo.SystemUUID
```

Comment:
Do you know the case there cloudproviderapi.AnnotationAlphaProvidedIPAddr is not set but ccmapi.TaintExternalCloudProvider taint exists?